idaes/models/properties/modular_properties/base/generic_property.py

```diff
@@ -3718,32 +3821,95 @@ def _therm_cond_phase(self):
         try:
 
             def rule_therm_cond_phase(b, p):
-                return get_phase_method(b, "therm_cond_phase", p)(b, p)
+                try:
+                    return get_phase_method(b, "therm_cond_phase", p)(b, p)
+                except GenericPropertyPackageError:
+                    # Handle case where thermal conductivity isn't defined for a phase
+                    return Expression.Skip
 
             self.therm_cond_phase = Expression(
                 self.phase_list,
-                doc="Thermal conductivity for each phase",
                 rule=rule_therm_cond_phase,
             )
         except AttributeError:
             self.del_component(self.therm_cond_phase)
             raise
 
+    def _therm_cond_phase_comp(self):
+        try:
+
+            def rule_therm_cond_phase_comp(b, p, j):
+                cobj = b.params.get_component(j)
+                if (
+                    cobj.config.therm_cond_phase_comp is not None
+                    and p in cobj.config.therm_cond_phase_comp
+                    and cobj.config.therm_cond_phase_comp[p] is not None
+                ):
+                    return cobj.config.therm_cond_phase_comp[
+                        p
+                    ].therm_cond_phase_comp.return_expression(b, cobj, p, b.temperature)
+                else:
+                    # Handle case where thermal conductivity isn't defined for a particular phase-component pair
+                    return Expression.Skip
```

Comment:
Do we really want to do this? This leaves a potential gap for users who might not be aware of this. My initially feeling is that if someone want to define a set of phase-component properties then they need to define all valid-phase component properties.

Reply:
Is this still necessary after the change to using `phase_component_set`? If not we should remove it, otherwise I think we need to have a discussion about what we should expect in these cases.

Reply:
I assume this is building off the conditional in `therm_cond_phase`. It is probably OK to assume that thermal conductivity is not defined for a specific phase, but my initial feeling is that it should be defined for all components within a given phase. If so, we should probably put a check in place for this.